### ISSUE TITLE
Doesn't work for actual big integers, parsing still truncates large numbers from JSON

### ISSUE THREAD

**Issue by Darker** [Ivan-Korolenko/json-with-bigint#7]
So this library made me look like a fool. I claimed to have fixed our issue with big ints by using this library, but I naively just trusted it works, which was a mistake. It does not work at all. It only converts one way, but the reverse still truncates the numbers.

All I have to do is add a single digit to the number from the example on the npm homepage:

```
import * as jsonBigInt from "json-with-bigint"

const data = { bigNumber: 90071992547409921n };
console.log("Data:   ", data.bigNumber);
const serialized = jsonBigInt.JSONStringify(data);
console.log("Serialized: ", serialized);
/** @type {typeof data} **/
const parsed = jsonBigInt.JSONParse(serialized);
console.log("Parsed: ", parsed.bigNumber);
console.log("Parsed (num to str): ", parsed.bigNumber.toString());
console.assert(parsed.bigNumber == data.bigNumber, "Numbers are not equal!");
```
Result:

```
Data:    90071992547409921n
Serialized:  {"bigNumber":90071992547409921}
Parsed:  90071992547409920
Parsed (num to str):  90071992547409920
Assertion failed: Numbers are not equal!
```

If this library is not meant to parse large numbers, it should say so front and center.

**Comment by Ivan-Korolenko** [comment]
@Darker Hi Jakub, 

Thank you for bringing this issue to my attention. The bug was introduced in one of the recent versions and only affected the ending BigInt-s in JSON, which is probably the reason why it took so much time for it to be discovered.

**This bug is fixed in version 2.1.3**. Feel free to download it via npm and see that the issue is fixed.

**Comment by Darker** [comment]
Thanks, that was really fast! Looking at the change though, this actually wouldn't work for me unfortunately. I am communicating with an external system that both sends and receives uint64 IDs, and will not accept string (I tried).

Since the other "json big int" library is also bugged, I wrote my own solution using the new `context` argument of JSON reviver and the new `JSON.rawJSON` for serializing back.
